packages/core/test/utils_test.js

```diff
@@ -3640,3 +3663,131 @@ describe("utils", () => {
     });
   });
 });
+
+describe("Utils.isCyclic", () => {
+  it("should catch inifinite recursion via circular referencing", () => {
+    const schema = {
+      type: "object",
+      definitions: {},
+      properties: {
+        foo: {
+          type: "object",
+          properties: {},
+        },
+      },
+    };
+    schema.properties.foo.properties.foo = schema.properties.foo;
+    const result = isCyclic(schema.properties.foo, undefined, schema);
+    expect(result).eql(true);
+  });
+
+  it("should catch infinite recursion via $ref", () => {
+    const schema = {
+      type: "object",
+      definitions: {
+        node: {
+          type: "object",
+          properties: {
+            otherNode: {
+              $ref: "#/definitions/node",
+            },
+          },
+        },
+      },
+      properties: {
+        tree: {
+          title: "Recursive references",
+          $ref: "#/definitions/node",
+        },
+      },
+    };
+    const result = isCyclic(schema, schema);
+    expect(result).eql(true);
+  });
+
+  it("should catch infinite recursion via $ref", () => {
+    const result = isCyclic(circularRef, circularRef);
+    expect(result).eql(true);
+  });
```

Comment:
This test isn't super realistic in the context of RJSF because RJSF will never receive a full OAS definition like this.

Reply:
Ok. I can take it out then